deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/ProcessorUtils.kt

```diff
@@ -0,0 +1,15 @@
+package com.airbnb.deeplinkdispatch
+
+object ProcessorUtils {
+    @JvmStatic
+    fun decapitalize(str: String?): String? {
+        return if (str == null || (str.length > 1 && str[1].isUpperCase() && str[0].isUpperCase())) {
+            str
+        } else {
+            str.replaceFirstChar { it.lowercase() }
+        }
+    }
+
+    @JvmStatic
+    fun hasEmptyOrNullString(strings: Array<String>) = strings.any { it.isNullOrEmpty() }
```

Comment:
Also nice as an extension function. And based on the name, should this be `Array<String?>`